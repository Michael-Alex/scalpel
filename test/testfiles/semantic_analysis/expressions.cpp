bool b1 = 7 < 9;
bool b2 = 7 <= 9;
bool b3 = 7 > 9;
bool b4 = 7 >= 9;
bool b5 = 7 == 9;
bool b6 = 7 != 9;

bool b7 = true && false;
bool b8 = true || false;
bool b9 = !true;

int i01 = 7 * 9;
int i02 = 7 / 9;
int i03 = 7 % 9;
int i04 = 7 + 9;
int i05 = 7 - 9;
int i06 = 7 << 2;
int i07 = 7 >> 2;

int i08 = 7 & 9;
int i09 = 7 ^ 9;
int i10 = 7 | 9;

int i11 = +7;
int i12 = -7;
int i13 = ~7;

int i14 = true ? 8 : 6;

