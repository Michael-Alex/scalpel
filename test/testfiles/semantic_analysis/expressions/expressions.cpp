bool a_bool = true;
int an_int = 7;

bool b1 = an_int < 9;
bool b2 = an_int <= 9;
bool b3 = an_int > 9;
bool b4 = an_int >= 9;
bool b5 = an_int == 9;
bool b6 = an_int != 9;

bool b7 = a_bool && false;
bool b8 = a_bool || false;
bool b9 = !a_bool;

int i01 = an_int * 9;
int i02 = an_int / 9;
int i03 = an_int % 9;
int i04 = an_int + 9;
int i05 = an_int - 9;
int i06 = an_int << 2;
int i07 = an_int >> 2;

int i08 = an_int & 9;
int i09 = an_int ^ 9;
int i10 = an_int | 9;

int i11 = +an_int;
int i12 = -an_int;
int i13 = ~an_int;

int i14 = a_bool ? an_int : 6;

