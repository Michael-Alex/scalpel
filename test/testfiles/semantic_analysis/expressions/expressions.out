{
  "id": 0,
  "variables":
  [
    {
      "id": 0,
      "name": "b1",
      "type":
      {
        "fundamental type": "bool"
      },
      "default value":
      {
        "less than":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 1,
      "name": "b2",
      "type":
      {
        "fundamental type": "bool"
      },
      "default value":
      {
        "less than or equal to":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 2,
      "name": "b3",
      "type":
      {
        "fundamental type": "bool"
      },
      "default value":
      {
        "greater than":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 3,
      "name": "b4",
      "type":
      {
        "fundamental type": "bool"
      },
      "default value":
      {
        "greater than or equal to":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 4,
      "name": "b5",
      "type":
      {
        "fundamental type": "bool"
      },
      "default value":
      {
        "equal to":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 5,
      "name": "b6",
      "type":
      {
        "fundamental type": "bool"
      },
      "default value":
      {
        "not equal to":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 6,
      "name": "b7",
      "type":
      {
        "fundamental type": "bool"
      },
      "default value":
      {
        "logical and":
        {
          "left operand":
          {
            "bool": true
          },
          "right operand":
          {
            "bool": false
          }
        }
      }
    },
    {
      "id": 7,
      "name": "b8",
      "type":
      {
        "fundamental type": "bool"
      },
      "default value":
      {
        "logical or":
        {
          "left operand":
          {
            "bool": true
          },
          "right operand":
          {
            "bool": false
          }
        }
      }
    },
    {
      "id": 8,
      "name": "b9",
      "type":
      {
        "fundamental type": "bool"
      },
      "default value":
      {
        "logical negation":
        {
          "operand":
          {
            "bool": true
          }
        }
      }
    },
    {
      "id": 9,
      "name": "i01",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "multiplication":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 10,
      "name": "i02",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "division":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 11,
      "name": "i03",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "modulo":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 12,
      "name": "i04",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "int": 16
      }
    },
    {
      "id": 13,
      "name": "i05",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "subtraction":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 14,
      "name": "i06",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "left shift":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 2
          }
        }
      }
    },
    {
      "id": 15,
      "name": "i07",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "right shift":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 2
          }
        }
      }
    },
    {
      "id": 16,
      "name": "i08",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "bitwise and":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 17,
      "name": "i09",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "bitwise exclusive or":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 18,
      "name": "i10",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "bitwise inclusive or":
        {
          "left operand":
          {
            "int": 7
          },
          "right operand":
          {
            "int": 9
          }
        }
      }
    },
    {
      "id": 19,
      "name": "i11",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "int": 7
      }
    },
    {
      "id": 20,
      "name": "i12",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "negation":
        {
          "operand":
          {
            "int": 7
          }
        }
      }
    },
    {
      "id": 21,
      "name": "i13",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "complement":
        {
          "operand":
          {
            "int": 7
          }
        }
      }
    },
    {
      "id": 22,
      "name": "i14",
      "type":
      {
        "fundamental type": "int"
      },
      "default value":
      {
        "conditional":
        {
          "condition operand":
          {
            "bool": true
          },
          "true operand":
          {
            "int": 8
          },
          "false operand":
          {
            "int": 6
          }
        }
      }
    }
  ]
}
